sdk/translation/ai-translation-text/samples-dev/translateProfanity.ts

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @summary This sample demonstrates how to change the profanity handling during translate call.
+ * Normally the Translator service will retain profanity that is present in the source
+ * in the translation. The degree of profanity and the context that makes words profane
+ * differ between cultures, and as a result the degree of profanity in the target language
+ * may be amplified or reduced.
+ * 
+ * If you want to avoid getting profanity in the translation, regardless of the presence
+ * of profanity in the source text, you can use the profanity filtering option. The option
+ * allows you to choose whether you want to see profanity deleted, whether you want to mark
+ * profanities with appropriate tags (giving you the option to add your own post-processing),
+ * or you want no action taken. The accepted values of `ProfanityAction` are `Deleted`, `Marked`
+ * and `NoAction` (default).
+ */
+import TextTranslationFactory, { ErrorResponseOutput, TranslatorCredential, InputTextItem, TranslateQueryParamProperties, TranslatedTextItemOutput } from "@azure-rest/ai-translation-text";
+
+import * as dotenv from "dotenv";
+dotenv.config();
+
+const endpoint = process.env["ENDPOINT"] || "https://api.cognitive.microsofttranslator.com";
+const apiKey = process.env["TEXT_TRANSLATOR_API_KEY"] || "<api key>";
+const region = process.env["TEXT_TRANSLATOR_REGION"] || "<region>";
+
+export async function main() {
+  console.log("== Profanity handling sample ==");
+
+  const translateCedential = new TranslatorCredential(apiKey, region);
+  const translationClient = TextTranslationFactory(endpoint, translateCedential, undefined);
+
+  const inputText: InputTextItem[] = [
+    { text: "This is ***." }
+  ];
+  const parameters: TranslateQueryParamProperties & Record<string, unknown> = {
```

Comment:
Why do we need the intersection with `Record<string, unknown>`?

Reply:
<img width="777" alt="image" src="https://user-images.githubusercontent.com/5716362/230253222-0f09e3e2-8b03-4a3b-b9ad-3c55ab375d26.png">

Reply:
I see, maybe you could get away with not typing the object?

Reply:
```typescript
  const parameters = {
    to: "cs",
    from: "en",
    profanityAction: "Marked",
    profanityMarker: "Asterisk"
  };
  const translateResponse = await translationClient.path("/translate").post({
    body: inputText,
    queryParameters: parameters
  })
```

<img width="906" alt="image" src="https://user-images.githubusercontent.com/5716362/230253969-1fb9abbd-f688-4a0c-a0f5-1ab45c05f46e.png">

Reply:
I have then issues with how the enums were generated:
![image](https://user-images.githubusercontent.com/5716362/230254212-e8300295-2566-4353-b0da-eb1effe98e82.png)

```typescript
  const parameters = {
    to: "cs",
    from: "en",
    profanityAction: "Marked",
    profanityMarker: "Asterisk"
  };
  const translateResponse = await translationClient.path("/translate").post({
    body: inputText,
    queryParameters: parameters
  })
```

Reply:
@joheredi That is a weird experience, should the object be marked as const?

Reply:
This may be an issue in our core library definition of queryParameters, I'll look into that. In the meantime I would recomend inlining the object 

```ts
  const translateResponse = await translationClient.path("/translate").post({
    body: inputText,
    queryParameters:  {
    to: "cs",
    from: "en",
    profanityAction: "Marked",
    profanityMarker: "Asterisk"
    }
  })
```

Reply:
We will wait for the investigation and hopefully by GA, we will be able to just use the correct type. No change in this PR.